src/interfaces/token.spec.ts

```diff
@@ -590,297 +589,97 @@ describe("TokenInterface", () => {
-      let vault: Asset<"VAULT_V2">;
-      let token: Address;
-
-      beforeEach(() => {
-        vault = createMockAssetStaticVaultV2();
-        token = createMockToken({ address: "0x999" }).address;
-        zapperGasMock.mockResolvedValue({
-          standard: 1,
-          instant: 2,
-          fast: 3
-        });
-      });
-
-      describe("when is not approved", () => {
-        beforeEach(() => {
-          zapperZapInApprovalStateMock.mockResolvedValue({
-            isApproved: false
-          });
-          zapperZapInApprovalTransactionMock.mockResolvedValue({
-            data: "data",
-            to: "0x000",
-            from: "0x001",
-            gasPrice: "1"
-          });
-        });
-
-        it("should approve vault to spend a token on zapIn", async () => {
-          const actualApprove = await tokenInterface.approveDeposit(vault.address, vault.token, token, "1", "0x001");
-
-          expect(actualApprove).toEqual("transaction");
-          expect(zapperZapInApprovalTransactionMock).toHaveBeenCalledTimes(1);
-          expect(zapperZapInApprovalTransactionMock).toHaveBeenCalledWith("0x001", "0x999", "3000000000", "yearn");
-        });
-      });
-
-      describe("when is approved", () => {
-        beforeEach(() => {
-          zapperZapInApprovalStateMock.mockResolvedValue({
-            isApproved: true
-          });
-        });
-
-        it("should return true", async () => {
-          const actualApprove = await tokenInterface.approveDeposit(vault.address, vault.token, token, "1", "0x001");
+    it("should throw when approving native token", async () => {
+      try {
+        await tokenInterface.approve(owner, ZeroAddress, spender, amount);
+      } catch (error) {
+        expect(error).toStrictEqual(new SdkError(`Native tokens cant be approved`));
+        expect(Contract).not.toHaveBeenCalled();
+        expect(approveMock).not.toHaveBeenCalled();
+        expect(sendTransactionMock).not.toHaveBeenCalled();
+      }
+    });
 
-          expect(actualApprove).toEqual(true);
-        });
-      });
+    it("should throw if approving token as its spender", async () => {
+      try {
+        await tokenInterface.approve(owner, spender, spender, amount);
+      } catch (error) {
+        expect(error).toStrictEqual(new SdkError(`Cant approve token as its spender`));
+        expect(Contract).not.toHaveBeenCalled();
+        expect(approveMock).not.toHaveBeenCalled();
+        expect(sendTransactionMock).not.toHaveBeenCalled();
+      }
     });
   });
 
   describe("allowance", () => {
-    describe("when the vault token is the same as the token", () => {
-      let vault: Asset<"VAULT_V2">;
-      let token: Address;
-
-      beforeEach(() => {
-        vault = createMockAssetStaticVaultV2();
-        token = createMockToken().address;
-        sendTransactionMock.mockResolvedValue(true);
-      });
+    const owner: Address = "0xOwner";
+    const spender: Address = "0xSpender";
```

Comment:
Maybe even here, or maybe it's just me that is not used to look at this and it should be obvious these are addresses rather than a `owner` object
```suggestion
    const ownerAddress: Address = "0xOwner";
    const spenderAddress: Address = "0xSpender";
```

Reply:
no, you are right. Renamed using more explicit names here https://github.com/yearn/yearn-sdk/pull/266/commits/5949fd4abc1094090e6bdd0f97a70bf07ee59604